src/plots/piePlot.ts

```diff
@@ -243,6 +267,89 @@ export module Plots {
+      var pointAngle = Math.acos(-p.y / (1 + pointRadius));
+      if (p.x < 0) {
+        pointAngle = Math.PI * 2 - pointAngle;
+      }
+      var index: number;
+      for (var i = 0; i < this._startAngles.length; i++) {
+        if (this._startAngles[i] < pointAngle && this._endAngles[i] > pointAngle) {
+          index = i;
+          break;
+        }
+      }
+      if (index !== undefined) {
+        var dataset = this.datasets()[0];
+        var datum = dataset.data()[index];
+        var innerRadius = this.innerRadius().accessor(datum, index, dataset);
+        var outerRadius = this.outerRadius().accessor(datum, index, dataset);
+        if (pointRadius > innerRadius && pointRadius < outerRadius) {
+          return index;
+        }
+      }
+      return null;
+    }
+
+    private _drawLabels() {
+      var attrToProjector = this._generateAttrToProjector();
+      var labelArea = this._renderArea.append("g").classed("label-area", true);
+      var measurer = new SVGTypewriter.Measurers.Measurer(labelArea);
+      var writer = new SVGTypewriter.Writers.Writer(measurer);
+      var dataset = this.datasets()[0];
+
+      for (var i = 0; i < dataset.data().length; i++) {
+        var datum = dataset.data()[i];
+        var value = "" + this.sectorValue().accessor(datum, i, dataset);
+        var measurement = measurer.measure(value);
+
+        var theta = (this._endAngles[i] + this._startAngles[i]) / 2;
+        var outerRadius = this.outerRadius().accessor(datum, i, dataset);
+        if (this.outerRadius().scale) {
+          outerRadius = this.outerRadius().scale.scale(outerRadius);
+        }
+        var innerRadius = this.innerRadius().accessor(datum, i, dataset);
+        if (this.innerRadius().scale) {
+          innerRadius = this.innerRadius().scale.scale(innerRadius);
+        }
+        var labelRadius = (outerRadius + innerRadius) / 2;
+
+        var x = Math.sin(theta) * labelRadius - measurement.width / 2;
+        var y = -Math.cos(theta) * labelRadius - measurement.height / 2;
+
+        var corners = [
+          { x: x, y: y },
+          { x: x, y: y + measurement.height},
+          { x: x + measurement.width, y: y },
+          { x: x + measurement.width, y: y + measurement.height }
+        ];
+
+        var sliceIndices = corners.map((corner) => this._sliceIndexForPoint(corner));
+        var showLabel = sliceIndices.every((index) => (index != null && index === sliceIndices[0]));
+
+        var color = attrToProjector["fill"](datum, i, dataset);
+        var dark = Utils.Color.contrast("white", color) * 1.6 < Utils.Color.contrast("black", color);
+        var g = labelArea.append("g").attr("transform", "translate(" + x + "," + y + ")");
+        var className = dark ? "dark-label" : "light-label";
+        g.classed(className, true);
+        g.style("display", showLabel ? "block" : "none");
+
+        writer.write(value, measurement.width, measurement.height, {
```

Comment:
Also some question as to whether we want to write the text at all if it's not visible. It's useful for debugging though, so I think we should keep doing it.